src/Stratis.Bitcoin/P2P/PeerAddress.cs

```diff
@@ -80,23 +80,58 @@ public IPAddress Loopback
         public DateTime? LastSeen { get; private set; }
 
         /// <summary>
-        /// The UTC Date when a Peer is banned.
-        /// <para>
-        /// This is set in <see cref="PeerBanning"/>, against <see cref="PeerAddress"/>.
-        /// </para>
+        /// UTC DateTime when a peer is banned.
         /// </summary>
-        [JsonProperty(PropertyName = "bandate", NullValueHandling = NullValueHandling.Ignore)]
-        public DateTime? BanDate { get; set; }
+        /// <remarks>
+        /// This is set in <see cref="PeerBanning"/>.
+        /// </remarks>
+        [JsonProperty(PropertyName = "bantimestamp", NullValueHandling = NullValueHandling.Ignore)]
+        public DateTime? BanTimeStamp { get; set; }
 
         /// <summary>
-        /// UTC date of when the ban will expire against a Peer.
-        /// <para>
-        /// This is set in <see cref="PeerBanning"/>, against <see cref="PeerAddress"/>.
-        /// </para>
+        /// UTC DateTime when the ban expires against the peer.
         /// </summary>
+        /// <remarks>
+        /// This is set in <see cref="PeerBanning"/>.
+        /// </remarks>
         [JsonProperty(PropertyName = "banuntil", NullValueHandling = NullValueHandling.Ignore)]
         public DateTime? BanUntil { get; set; }
 
+        /// <summary>
+        /// Reason for banning the peer.
+        /// </summary>
+        /// <remarks>
+        /// This is set in <see cref="PeerBanning"/>.
+        /// </remarks>
+        [JsonProperty(PropertyName = "banreason", NullValueHandling = NullValueHandling.Ignore)]
+        public string BanReason { get; set; }
+
+        /// <summary>
+        /// Maintain a count of bad behaviour.  
+        /// </summary>
+        /// <para>
+        /// Once a certain score is reached ban the peer.
+        /// </para>
+        /// <remarks>
+        /// The logic around this has not yet been implemented.
+        /// This is set in <see cref="PeerBanning"/>.
+        /// </remarks>
+        [JsonProperty(PropertyName = "banscore", NullValueHandling = NullValueHandling.Ignore)]
+        public uint? BanScore { get; set; }
+
+        /// <summary>
+        /// <c>True</c> if <see cref="BanUntil"/> is in the future.
+        /// </summary>
+        [JsonIgnore]
+        public bool IsBanned
+        {
+            get
+            {
+                return 
+                    (this.BanUntil > DateTime.UtcNow.Date);
```

Comment:
Remove this method, because otherwise you would need date time provider to be injected. Never use date time UTC now directly.